packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -687,7 +687,8 @@ function createInstances(cluster: DatabaseClusterNew, props: DatabaseClusterBase
       props.clusterIdentifier != null ? `${props.clusterIdentifier}instance${instanceIndex}` :
         undefined;
 
-    const publiclyAccessible = instanceProps.vpcSubnets && instanceProps.vpcSubnets.subnetType === ec2.SubnetType.PUBLIC;
+    const publiclyAccessible = instanceProps.publiclyAccessible ?? (instanceProps.vpcSubnets &&
+    instanceProps.vpcSubnets.subnetType === ec2.SubnetType.PUBLIC);
```

Comment:
This formatting is a little weird... how about

```suggestion
    const publiclyAccessible = instanceProps.publiclyAccessible ??
      (instanceProps.vpcSubnets && instanceProps.vpcSubnets.subnetType === ec2.SubnetType.PUBLIC);
```